src/main/java/duke/storage/DukeStorage.java

```diff
@@ -0,0 +1,126 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+// Class that handles the loading and saving of the save file.
+public class DukeStorage implements Storage {
+    private final Path filePath;
+    private ArrayList<String> saveLines;
+    private boolean isActive = true;
+
+    /**
+     * Initialises the DukeStorage object.
+     * Creates the save file in the specified path if it does not exists
+     * then loads the save file.
+     *
+     * @param filePath Path path to save file.
+     * @throws IOException If an IO error occurs while creating or loading the save file.
+     */
+    public DukeStorage(Path filePath) throws IOException {
+        this.filePath = filePath;
+        try {
+            createIfNotExist();
```

Comment:
I don't think this is grammatically correct. However, the more crucial point is regarding the method name: What is the subject being referred to here, i.e. create if _what_ does not exist? (to consider renaming)